### PR TITLE
Feature/chat improvements noti

### PR DESCRIPTION
**Pull request by kasenvr**
In essence this...

- Adds a local notification for people who enter or leave a domain.
- Makes the chat window dialog open when you click on a chat message notification box at the bottom of the screen.
- Updates license headers and fills in a missing one.

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-b904612_22085e21.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-b904612.tar_22085e21.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-b904612.tar_22085e21.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-b904612_22085e21.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-b904612_22085e21.exe

**Comment by Aitolda** [comment]
Seems to be working.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-96afbc2_431f1886.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-96afbc2_431f1886.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-96afbc2.tar_431f1886.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-96afbc2.tar_431f1886.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-96afbc2_431f1886.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-0379be6_b64c44f5.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-0379be6.tar_b64c44f5.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-0379be6.tar_b64c44f5.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-0379be6_b64c44f5.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-0379be6_b64c44f5.exe

**Comment by Aitolda** [comment]
Update with mute for notices works as expected.

File: scripts/communityScripts/chat/FloofChat.js [hunk 1]
```diff
@@ -618,6 +625,46 @@ function setVisible(_visible) {
     visible = _visible;
 }
 
+function avatarJoinsDomain(sessionID) {
+    Script.setTimeout(function () {
+        var messageText = AvatarManager.getPalData([sessionID]).data[0].sessionDisplayName + " has joined."
+        var messageColor = {red: 122, green: 122, blue: 122};
```
Comment (ctrlaltdavid) [inline scripts/communityScripts/chat/FloofChat.js:631]:
```suggestion
        var messageColor = { red: 122, green: 122, blue: 122 };
```

File: scripts/communityScripts/chat/FloofChat.js [hunk 2]
```diff
@@ -618,6 +625,46 @@ function setVisible(_visible) {
...
+        
+        if (!mutedAudio["Domain"]) {
+            playNotificationSound();
+        }
+        
+        if (!muted["Domain"]) {
+            Messages.sendLocalMessage(FLOOF_NOTIFICATION_CHANNEL, JSON.stringify({
+                sender: "(D)",
+                text:  messageText,
+                colour: {text: messageColor}
```
Comment (ctrlaltdavid) [inline scripts/communityScripts/chat/FloofChat.js:643]:
```suggestion
                colour: { text: messageColor }
```

File: scripts/communityScripts/chat/FloofChat.js [hunk 3]
```diff
@@ -618,6 +625,46 @@ function setVisible(_visible) {
...
+            }));
+        }
+    }, 500); // Wait 500ms for the avatar to load to properly get info about them.
+}
+
+function avatarLeavesDomain(sessionID) {
+    var messageText = AvatarManager.getPalData([sessionID]).data[0].sessionDisplayName + " has left."
+    var messageColor = {red: 122, green: 122, blue: 122};
```
Comment (ctrlaltdavid) [inline scripts/communityScripts/chat/FloofChat.js:651]:
```suggestion
    var messageColor = { red: 122, green: 122, blue: 122 };
```

File: scripts/communityScripts/chat/FloofChat.js [hunk 4]
```diff
@@ -618,6 +625,46 @@ function setVisible(_visible) {
...
+    
+    if (!mutedAudio["Domain"]) {
+        playNotificationSound();
+    }
+    
+    if (!muted["Domain"]) {
+        Messages.sendLocalMessage(FLOOF_NOTIFICATION_CHANNEL, JSON.stringify({
+            sender: "(D)",
+            text:  messageText,
+            colour: {text: messageColor}
```
Comment (ctrlaltdavid) [inline scripts/communityScripts/chat/FloofChat.js:663]:
```suggestion
            colour: { text: messageColor }
```

File: scripts/communityScripts/notificationCore/notificationCore.js [hunk 5]
```diff
@@ -161,4 +176,19 @@ function notif(text, colour) {
     notificationList.push(noti);
 }
 
+Controller.mousePressEvent.connect(function (event) {
+    // Overlays.getOverlayAtPoint applies only to 2D overlays.
+    var overlay = Overlays.getOverlayAtPoint({ x: event.x, y: event.y });
+    if (overlay) {
+        for (var i = 0; i < notificationList.length; i++) {
+            if (overlay == notificationList[i].id) {
```
Comment (ctrlaltdavid) [inline scripts/communityScripts/notificationCore/notificationCore.js:184]:
```suggestion
            if (overlay === notificationList[i].id) {
```

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-bab7ae1_cc43112b.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-bab7ae1.tar_cc43112b.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-bab7ae1_cc43112b.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-bab7ae1.tar_cc43112b.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR579/Vircadia-Alpha-PR579-bab7ae1_cc43112b.exe

**Pull request merged**